persistence-modules/spring-data-elasticsearch/src/test/java/com/baeldung/elasticsearch/GeoQueriesManualTest.java

```diff
@@ -1,199 +1,192 @@
+                        .properties("region", region -> region.geoShape(gs -> gs))
+                        .properties("location", location -> location.geoPoint(gp -> gp))
+                )
+        );
     }
 
-//    @Test
-//    public void givenGeoShapeData_whenExecutedGeoShapeQuery_thenResultNonEmpty() throws IOException {
-//        String jsonObject = "{\"name\":\"Agra\",\"region\":{\"type\":\"envelope\",\"coordinates\":[[75,30.2],[80.1, 25]]}}";
-//        IndexRequest indexRequest = new IndexRequest(WONDERS_OF_WORLD);
-//        indexRequest.source(jsonObject, XContentType.JSON);
-//        IndexResponse response = client.index(indexRequest, RequestOptions.DEFAULT);
-//
-//        String tajMahalId = response.getId();
-//
-//        RefreshRequest refreshRequest = new RefreshRequest(WONDERS_OF_WORLD);
-//        client.indices()
-//            .refresh(refreshRequest, RequestOptions.DEFAULT);
-//
-//        Coordinate topLeft = new Coordinate(74, 31.2);
-//        Coordinate bottomRight = new Coordinate(81.1, 24);
-//
-//        GeoShapeQueryBuilder qb = QueryBuilders.geoShapeQuery("region", new EnvelopeBuilder(topLeft, bottomRight).buildGeometry());
-//        qb.relation(ShapeRelation.INTERSECTS);
-//
-//        SearchSourceBuilder source = new SearchSourceBuilder().query(qb);
-//        SearchRequest searchRequest = new SearchRequest(WONDERS_OF_WORLD);
-//        searchRequest.source(source);
-//
-//        SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
-//
-//        List<String> ids = Arrays.stream(searchResponse.getHits()
-//            .getHits())
-//            .map(SearchHit::getId)
-//            .collect(Collectors.toList());
-//
-//        assertTrue(ids.contains(tajMahalId));
-//    }
+    @Test
+    public void givenGeoShapeData_whenExecutedGeoShapeQuery_thenResultNonEmpty() throws IOException {
+        String jsonObject = """
+                {
+                    "name":"Agra",
+                    "region":{
+                        "type":"envelope",
+                        "coordinates":[[75,30.2],[80.1,25]]
+                    }
+                }
+                """;
+        IndexResponse response = client.index(idx -> idx
+                .index(WONDERS_OF_WORLD)
+                .withJson(new StringReader(jsonObject)));
+
+        String tajMahalId = response.id();
+        client.indices().refresh();
+
+        StringReader jsonData = new StringReader("""
+                {
+                     "type":"envelope",
+                     "coordinates": [[74.0, 31.2], [81.1, 24.0 ] ]
+                }
+                """);
+
+        SearchRequest searchRequest = new SearchRequest.Builder()
+                .query(qr -> qr.bool(b -> b.filter(f -> f.geoShape(gs -> gs.field("region")
+                        .shape(shp -> shp.relation(GeoShapeRelation.Within)
+                                .shape(JsonData.from(jsonData))
+                        ))))).build();
```

Comment:
I think this statement needs to use complete variable names and different indentation to improve readability.